Update security description (#97)
sourcegraph/careers: job-descriptions/software-engineer-security.md

# Software Engineer - Security

> We are looking for an expert to own security at Sourcegraph. This includes the security of Sourcegraph.com as well as making it easy for our customers to secure their own private Sourcegraph instances that they deploy on-premise. You will have the opportunity to build a world-class security culture and team from the ground up.

Reviewer: ```suggestion
We are looking for an expert to own security at Sourcegraph. This includes the security of Sourcegraph Cloud as well as making it easy for our customers to secure their own private Sourcegraph instances that they deploy on-premise. You will have the opportunity to build a world-class security culture and team from the ground up.
```

> We are looking for an expert to own security at Sourcegraph. This includes the security of Sourcegraph.com as well as making it easy for our customers to secure their own private Sourcegraph instances that they deploy on-premise. You will have the opportunity to build the best security culture and organization in the world from the ground up.

Reviewer 2: ```suggestion
We are looking for an expert to own security at Sourcegraph. This includes the security of Sourcegraph.com as well as making it easy for our customers to secure their own private Sourcegraph instances that they deploy on-premise. You will have the opportunity to establish and build a world-class security and organizational practices from the ground up.
```

Current wording sounds a bit far out there; we aren't a security firm so how could we possibly be _the_ best? :)

Author: Hah, my first draft used the words “world class” and then I made it more aggressive because why not, this is supposed to be inspirational :)

I’ll think about it.

Author: ```suggestion
We are looking for an expert to own security at Sourcegraph. This includes the security of Sourcegraph.com as well as making it easy for our customers to secure their own private Sourcegraph instances that they deploy on-premise. You will have the opportunity to build a world-class security culture and team from the ground up.
```

## Responsibilities

In the first 30 days, you will:

- Identify what our security needs are and develop a roadmap to improve and ensure the security of our product.
- Promptly respond to and address security issues reported to us by third parties.

In the first 60 days, you will:

> - Execute on that roadmap through a mix of directly contributing to our codebase (i.e., Go, TypeScript, Kubernetes, Docker, Google Cloud Platform) yourself, and helping other engineers on our team make the necessary changes. You should expect that at least 75% of your effort will be direct contributions to start.

Reviewer: ```suggestion
- Execute on that roadmap through a mix of directly contributing to our codebase (i.e., Go, TypeScript, Kubernetes, Docker, Google Cloud Platform), and helping other engineers on our team make the necessary changes. You should expect that at least 75% of your effort will be direct contributions to start.
```

- Create a hiring plan and interview process to build the team that is necessary to secure Sourcegraph.

In the first year, you will:

> - Be the engineering manager for the security team that you have built.

Reviewer 2: It seems a bit of a stretch that we will find someone who can pull off *both* 75% direct contributions AND be a strong manager. I almost feel like these two are in direct opposition to one another.

I would propose rephrasing it to something like "75% direct contributions + be the team lead (and be willing to perform engineering manager duties until we find an appropriate manager)" or something?

Author: Yeah, I agree. @tsenart is going to manage the Sourcegraph Cloud project so we really don’t need another manager as soon as this plan implies. I need to think about how to reword this.

Author: I am actually not sure I want to concede this point. I don't think it is necessarily a stretch because we don't need someone to do both simultaneously. First we need an IC, then we will need an engineering manager at some point. In an ideal world this would be the same person and I want this job description to describe our ideal hire. This is similar to Christina's (first PM) and Rob's (first Designer) path at Sourcegraph. We need the equivalent for security.

I am curious to hear what @tsenart thinks.

Reviewer 2: I agree we need the equivalent here at Sourcegraph 100%. But, did we indicate to Christina or Rob that they would be the people manager of the entire design team or people manager of all future product managers? I didn’t think we did

In my mind, if we found somebody who had all of these qualifications and was a strong technical lead for security but was not interested or does not have strong people management skills they would still be a good fit for this role. Are you thinking different, or are we debating wording semantics only?

Reviewer 3: > First we need an IC, then we will need an engineering manager at some point.

Organisationally, how are we envisioning this to be? I think having security engineers embedded in mission teams (e.g. Sourcegraph Cloud) makes sense at first, like it makes sense to have other kinds of engineers embedded and having the mission team's engineering manager be their manager.

On the other hand, I am not 100% convinced that security lends itself to this organisational model as well as other kinds of engineering — while we want a culture of every team owning the security of what they build and run, it is critical that we devise working mechanisms of consistency to our security approach.

Having a centralized team makes that needed consistency easier, but, leads to reduced individual team flexibility on some fronts, longer decision making and reviewing processes, etc. This may be a required growing pain — I think if we are to host private customer code on Sourcegraph Cloud, we will have to, over time, overhaul our internal processes to be more restrictive and secure (and slower, comparatively) vs flexible and quick.

Overall, juggling this trade-off will be one of the key challenges we'll face organisationally, as it manifests in other maturing functions too (e.g. Ops).

For now, I'm happy managing a security tech lead that owns security at large and working on the most immediate needs in the context of Sourcegraph Cloud. But eventually, we need to answer the questions of organizational design, and it would be great for us to learn more about how similar companies have evolved over different stages.

Reviewer 3: To be clear, I'm also fine with us already pursuing the path of building a centralized security team, and finding a suitable leader for it now — I think that person would be able to work effectively in the Sourcegraph Cloud project as an IC / tech lead, given appropriate context, despite their medium to longer term goals around building a security team (manager).

Author: I am removing this requirement because I do think we need to hire a hands-on IC for today. Potential future management skills are a nice to have.
